cogito/putter.go

```diff
@@ -51,22 +51,42 @@ func (putter *ProdPutter) LoadConfiguration(input []byte, args []string) error {
 		"environment", putter.Request.Env,
 		"args", args)
 
+	var sinks *sets.Set[string]
+	// if put.params.sinks is not empty it overrides source.params.sinks.
+	if len(putter.Request.Params.Sinks) > 0 {
+		sinks = sets.From(putter.Request.Params.Sinks...)
+	} else {
+		sinks = sets.From(putter.Request.Source.Sinks...)
+	}
+
+	// Validate optional sinks configuration.
+	err = validateSinks(sinks)
+	if err != nil {
+		return fmt.Errorf("put: arguments: unsupported sink(s): %w", err)
+	}
+
+	// If no configured sinks or magic word 'github' is found, input dir is mandatory.
+	required := sinks.Size() == 0 || sinks.Contains("github")
+
 	// args[0] contains the path to a directory containing all the "put inputs".
-	if len(args) == 0 {
+	if len(args) == 0 && required {
 		return fmt.Errorf("put: arguments: missing input directory")
 	}
-	putter.InputDir = args[0]
-	putter.log.Debug("", "input-directory", putter.InputDir)
-
+	if len(args) > 0 {
+		putter.InputDir = args[0]
+		putter.log.Debug("", "input-directory", putter.InputDir)
+	}
```

Comment:
>Said otherwise, having len(args) == 0 is a concourse resource protocol violation error

This a part of the code that I didn't touch, but I agree with you, I'll update.

Reply:
I think that `required` has been added in this PR. In any case, good simplification, thanks Olivier.